### PR TITLE
API fix for REST access

### PR DESCRIPTION
**Pull request by sayan-biswas**
This PR addresses the following [issue-318](https://github.com/tektoncd/results/issues/318)

**Fix**
- `logs` static path fragment added to differentiate between `GetRecords` and `GetLogs` in the generated HTTP mux, keeping the input field `name` same.
- `ListLogs` request/response changed to print the whole log record (similar to `ListRecords`)

**Note** : Merge this before [PR-301](https://github.com/tektoncd/results/pull/301)

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/ok-to-test

File: proto/v1alpha2/api.proto [hunk 1]
```diff
@@ -96,14 +96,14 @@ service Results {
 service Logs {
   rpc GetLog(GetLogRequest) returns (stream Log) {
     option (google.api.http) = {
-      get: "/v1alpha2/{name=*/results/*/logs/*}"
+      get: "/v1alpha2/logs/{name=*/results/*/records/*}"
     };
```
Comment (khrm) [inline proto/v1alpha2/api.proto:100]: Please add `results` prefix for Results API.

Also, won't it be better to have rest APIs like below: 
`/apis/results.tekton.dev/v1alpha2/logs`

Comment (sayan-biswas) [reply proto/v1alpha2/api.proto:100]: Right.. I think this `/apis/results.tekton.dev/v1alpha2/logs` pattern is better. This follows the kube api access pattern. 
I'll change this for all the RPCs

Comment (avinal) [reply proto/v1alpha2/api.proto:100]: +1

**Review by khrm — APPROVED** [review]

**Review by adambkaplan — APPROVED** [review]
/approve

@sayan-biswas can we currently test this end to end? Or do we need to put more effort in (example - get tkn-results in shape to use as a client).

**Comment by khrm** [comment]
We can test this using CLI `grpc_cli` call. That's how I did it.

**Comment by khrm** [comment]
We can test this using CLI grpc_cli call. That's how I tested it, @adambkaplan .

**Comment by khrm** [comment]
Sorry. I thought you were talking about different pr. No, I didn't test this.

**Review by avinal — APPROVED** [review]
Hi @adambkaplan IIRC we don't have any test yet for testing the Rest APIs e2e, so we can only manually test it for now. Here are some test results from my test.
image


Log API implementation is due in #301 
image

**Comment by sayan-biswas** [comment]
> /approve
> 
> @sayan-biswas can we currently test this end to end? Or do we need to put more effort in (example - get tkn-results in shape to use as a client).

As of now, we can test e2e for Results and Records APIs, with `curl`. For logs API we need the server and watcher changes merged.

**Review by adambkaplan — COMMENTED** [review]
I'm actually going to propose that we use `parents` as the leading field in the API. This will make the Results API similar to k8s, where `namespaces` leads namespace-scoped paths in the REST API [1]. This will also eliminate the semi-stutter we have with the `results` resource type.

```
/apis/results.tekton.dev/v1alpha2/parents/{parent=*}`
```

@dibyom @alan-ghelardi what do you think?

[1] https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-uris

**Comment by alan-ghelardi** [comment]
Sounds good to me.

**Comment by sayan-biswas** [comment]
@adambkaplan The parent field stores `namespace` name. So can we call it namespace in the path?
For  results/records
`/apis/results.tekton.dev/v1alpha2/namespaces/{name=*/results/*}`
`/apis/results.tekton.dev/v1alpha2/namespaces/{name=*/results/*/records/*}`

For logs
`/apis/results.tekton.dev/v1alpha2/namespaces/{name=*/results/*/logs/*}`

**Comment by adambkaplan** [comment]
> @adambkaplan The parent field stores `namespace` name. So can we call it namespace in the path? For results/records `/apis/results.tekton.dev/v1alpha2/namespaces/{name=*/results/*}` `/apis/results.tekton.dev/v1alpha2/namespaces/{name=*/results/*/records/*}`
> 
> For logs `/apis/results.tekton.dev/v1alpha2/namespaces/{name=*/results/*/logs/*}`

Conceptually Results has the idea of a "parent" which is independent of Kubernetes. It _can_ be, and often is, a k8s namespace. But I believe the design intent is to allow for "other things" to be a parent (example - a git commit sha). @dibyom  and @wlynch may be able to shed more light here. Therefore I'm reluctant to tie the API completely to k8s REST conventions.

In practice - especially if RBAC-based auth is enabled - the parent is a namespace.

**Comment by dibyom** [comment]
> Conceptually Results has the idea of a "parent" which is independent of Kubernetes. It can be, and often is, a k8s namespace. But I believe the design intent is to allow for "other things" to be a parent (example - a git commit sha). @dibyom and @wlynch may be able to shed more light here. Therefore I'm reluctant to tie the API completely to k8s REST conventions.

Yup! @adambkaplan is right - for instance if you were to share the same results API across multiple tekton clusters, the parent cloud be cluster/<cluster-id>/namespace/<ns>/results/<id>

https://github.com/tektoncd/community/blob/main/teps/0021-results-api.md#result-api has some more details!

**Comment by sayan-biswas** [comment]
@adambkaplan @dibyom That makes sense, I'll go ahead with the `parents/{parent=*}` path.

**Comment by adambkaplan** [comment]
Can confirm that the URL path is resolving as expected. The x509 error is an artifact of the e2e script not adding "localhost" as a CN:

```sh
$ curl --insecure https://localhost:8080/apis/results.tekton.dev/v1alpha2/parents/example-build/results
{"code":14, "message":"connection error: desc = \"trans    port: authentication handshake failed: x509: certificat    e is valid for tekton-results-api-service.tekton-pipeli    nes.svc.cluster.local, not localhost\"", "details":[]}
```

**Comment by adambkaplan** [comment]
Hrm, something isn't right. When I try to curl the REST service (fixed separately), I consistently get "not found" responses:

```
$ kubectl get taskrun -n example-build -o yaml 
apiVersion: tekton.dev/v1beta1
kind: TaskRun
metadata:
  annotations:
    results.tekton.dev/record: example-build/results/6ae44060-0976-4796-be08-a3f4d8230eb3/records/6ae44060-0976-4796-be08-a3f4d8230eb3
    results.tekton.dev/result: example-build/results/6ae44060-0976-4796-be08-a3f4d8230eb3
...
$ curl -k https://localhost:8080/apis/results.tekton.dev/parents/example-build/results/6ae44060-0976-4796-be08-a3f4d8230eb3
{"code":5, "message":"Not Found", "details":[]}
```

**Comment by sayan-biswas** [comment]
@adambkaplan. The api version `v1alpha2` is missing in the curl URI. Can you try with that once?

**Comment by sayan-biswas** [comment]
Some results from my testing.. 

GetRecord
image

GetLog
image

**Review by khrm — COMMENTED** [review]
/kind bug
We need this label for release as discussed in the last WG call.

**Pull request merged**